File: backend/src/main/java/edu/cit/chan/unilost/features/claim/ClaimDTO.java
package edu.cit.chan.unilost.features.claim;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ClaimDTO {

    private String id;
    private String status;
    private String providedAnswer;
    private String message;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    // Resolved item info
    private String itemId;
    private String itemTitle;
    private String itemType;
    private String itemImageUrl;

    // Resolved claimant info
    private String claimantId;
    private String claimantName;
    private String claimantSchool;

    // Resolved finder info
    private String finderId;
    private String finderName;

    // Only populated for finder/admin views
    private String secretDetailQuestion;

    // Chat room associated with this claim
    private String chatId;

    // Handover tracking
    private LocalDateTime finderMarkedReturnedAt;
    private LocalDateTime ownerConfirmedReceivedAt;
}

File: mobile/app/src/main/java/com/hulampay/mobile/data/model/ClaimDto.kt
package com.hulampay.mobile.data.model

import com.google.gson.annotations.SerializedName

/**
 * Mirrors backend ClaimDTO returned from the /api/claims endpoints.
 *
 * For FOUND items the `finderId` is the item poster; for LOST items the `finderId`
 * is the claimant themselves. The handover timestamps drive the state machine:
 * `finderMarkedReturnedAt` -> item PENDING_OWNER_CONFIRMATION,
 * `ownerConfirmedReceivedAt` -> claim COMPLETED, item RETURNED.
 */
data class ClaimDto(
    @SerializedName("id")             val id: String = "",
    @SerializedName("status")         val status: String = "",
    @SerializedName("providedAnswer") val providedAnswer: String? = null,
    @SerializedName("message")        val message: String = "",
    @SerializedName("createdAt")      val createdAt: String? = null,
    @SerializedName("updatedAt")      val updatedAt: String? = null,

    @SerializedName("itemId")         val itemId: String = "",
    @SerializedName("itemTitle")      val itemTitle: String = "",
    @SerializedName("itemType")       val itemType: String = "",
    @SerializedName("itemImageUrl")   val itemImageUrl: String? = null,

    @SerializedName("claimantId")     val claimantId: String = "",
    @SerializedName("claimantName")   val claimantName: String = "",
    @SerializedName("claimantSchool") val claimantSchool: String? = null,

    @SerializedName("finderId")       val finderId: String = "",
    @SerializedName("finderName")     val finderName: String = "",

    @SerializedName("secretDetailQuestion")    val secretDetailQuestion: String? = null,
    @SerializedName("chatId")                  val chatId: String? = null,
    @SerializedName("finderMarkedReturnedAt")  val finderMarkedReturnedAt: String? = null,
    @SerializedName("ownerConfirmedReceivedAt") val ownerConfirmedReceivedAt: String? = null,
)
